include/bluetooth/services/ras.h

```diff
@@ -0,0 +1,138 @@
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+
+/** @file
+ *  @defgroup bt_ras Ranging Service API
+ *  @{
+ *  @brief API for the Ranging Service (RAS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief UUID of the Ranging Service. **/
+#define BT_UUID_RANGING_SERVICE_VAL (0x185B)
+
+/** @brief UUID of the RAS Features Characteristic. **/
+#define BT_UUID_RAS_FEATURES_VAL (0x2C14)
+
+/** @brief UUID of the Real-time Ranging Data Characteristic. **/
+#define BT_UUID_RAS_REALTIME_RD_VAL (0x2C15)
+
+/** @brief UUID of the On-demand Ranging Data Characteristic. **/
+#define BT_UUID_RAS_ONDEMAND_RD_VAL (0x2C16)
+
+/** @brief UUID of the RAS Control Point Characteristic. **/
+#define BT_UUID_RAS_CP_VAL (0x2C17)
+
+/** @brief UUID of the Ranging Data Ready Characteristic. **/
+#define BT_UUID_RAS_RD_READY_VAL (0x2C18)
+
+/** @brief UUID of the Ranging Data Overwritten Characteristic. **/
+#define BT_UUID_RAS_RD_OVERWRITTEN_VAL (0x2C19)
+
+#define BT_UUID_RANGING_SERVICE    BT_UUID_DECLARE_16(BT_UUID_RANGING_SERVICE_VAL)
+#define BT_UUID_RAS_FEATURES       BT_UUID_DECLARE_16(BT_UUID_RAS_FEATURES_VAL)
+#define BT_UUID_RAS_REALTIME_RD    BT_UUID_DECLARE_16(BT_UUID_RAS_REALTIME_RD_VAL)
+#define BT_UUID_RAS_ONDEMAND_RD    BT_UUID_DECLARE_16(BT_UUID_RAS_ONDEMAND_RD_VAL)
+#define BT_UUID_RAS_CP             BT_UUID_DECLARE_16(BT_UUID_RAS_CP_VAL)
+#define BT_UUID_RAS_RD_READY       BT_UUID_DECLARE_16(BT_UUID_RAS_RD_READY_VAL)
+#define BT_UUID_RAS_RD_OVERWRITTEN BT_UUID_DECLARE_16(BT_UUID_RAS_RD_OVERWRITTEN_VAL)
+
+#define BT_RAS_RANGING_HEADER_LEN  4
+#define BT_RAS_SUBEVENT_HEADER_LEN 8
+#define BT_RAS_STEP_MODE_LEN       1
+#define BT_RAS_MAX_STEP_DATA_LEN   35
+
+/** @brief Ranging Header structure as defined in RAS Specification, Table 3.7. */
+struct ras_ranging_header {
+	/** Ranging Counter is lower 12-bits of CS Procedure_Counter provided by the Core Controller
+	 *  (Core Specification, Volume 4, Part E, Section 7.7.65.44).
+	 */
+	uint16_t ranging_counter : 12;
+	/** CS configuration identifier. Range: 0 to 3. */
+	uint8_t  config_id       : 4;
+	/** Transmit power level used for the CS Procedure. Range: -127 to 20. Units: dBm. */
+	int8_t   selected_tx_power;
+	/** Antenna paths that are reported:
+	 *  Bit0: 1 if Antenna Path_1 included; 0 if not.
+	 *  Bit1: 1 if Antenna Path_2 included; 0 if not.
+	 *  Bit2: 1 if Antenna Path_3 included; 0 if not.
+	 *  Bit3: 1 if Antenna Path_4 included; 0 if not.
+	 *  Bits 4-7: RFU
+	 */
+	uint8_t  antenna_paths_mask;
+} __packed;
+BUILD_ASSERT(sizeof(struct ras_ranging_header) == BT_RAS_RANGING_HEADER_LEN);
+
+/** @brief Subevent Header structure as defined in RAS Specification, Table 3.8. */
+struct ras_subevent_header {
+	/** Starting ACL connection event count for the results reported in the event */
+	uint16_t start_acl_conn_event;
+	/** Frequency compensation value in units of 0.01 ppm (15-bit signed integer) */
```

Comment:
here if possible I would suggest referencing BT_HCI_LE_CS_SUBEVENT_RESULT_FREQ_COMPENSATION_NOT_AVAILABLE (or mentioning that 0xC000 is a special value). Otherwise I think the user might not realize they need to check for this value before converting this to a signed integer